tests/test_publish_dafsa.py

```diff
@@ -0,0 +1,191 @@
+        self.client = Client(
+            server_url=server, auth=auth, bucket=BUCKET_ID, collection=COLLECTION_ID
+        )
+        self.record_uri = server + self.client.get_endpoint(
+            "record", id=RECORD_ID, bucket=BUCKET_ID, collection=COLLECTION_ID
+        )
+
+    @responses.activate
+    def test_KintoException_raised_when_stored_hash_fetching_failed(self):
+        responses.add(
+            responses.GET, self.record_uri, json={"error": "not found"}, status=404
+        )
+        with self.assertRaises(KintoException) as e:
+            stored_hash = get_stored_hash(self.client)  # noqa
+            self.assertEqual(e.status_code, 404)
+
+
+class TestPrepareDafsa(unittest.TestCase):
+    def test_file_is_created_in_output_folder(self):
+        with tempfile.TemporaryDirectory() as tmp:
+            output_binary_path = prepare_dafsa(tmp)
+            self.assertIn(os.path.basename(output_binary_path), os.listdir(tmp))
+            self.assertGreater(os.path.getsize(output_binary_path), 0)
+
+    def test_exception_is_raised_when_process_returns_non_zero(self):
+        with tempfile.TemporaryDirectory() as tmp:
+            with mock.patch("subprocess.Popen") as mocked:
+                mocked.return_value.returncode = 1
+                with self.assertRaises(Exception) as e:
+                    prepare_dafsa(tmp)
+                    self.assertIn("DAFSA Build Failed", str(e.exception))
+
+
+class TestRemoteSettingsPublish(unittest.TestCase):
+    def setUp(self):
+        server = "https://fake-server.net/v1"
+        auth = ("arpit73", "pAsSwErD")
+        self.client = Client(
+            server_url=server, auth=auth, bucket=BUCKET_ID, collection=COLLECTION_ID
+        )
+        record_uri = server + self.client.get_endpoint(
+            "record", id=RECORD_ID, bucket=BUCKET_ID, collection=COLLECTION_ID
+        )
+        self.collection_uri = server + self.client.get_endpoint(
+            "collection", bucket=BUCKET_ID, collection=COLLECTION_ID
+        )
+        self.attachment_uri = f"{record_uri}/attachment"
+
+    @responses.activate
+    def test_record_was_posted(self):
+        responses.add(
+            responses.POST,
+            self.attachment_uri,
+            # body={"attachment": ("dafsa.bin", b"some binary data")},
+            json={"commit-hash": "fake-commit-hash"},
+        )
+        responses.add(
+            responses.PATCH, self.collection_uri, json={"data": {"status": "to-review"}}
+        )
+
+        with tempfile.TemporaryDirectory() as tmp:
+            dafsa_filename = f"{tmp}/dafsa.bin"
+            with open(dafsa_filename, "wb") as f:
+                f.write(b"some binary data")
+            remote_settings_publish(self.client, "fake-commit-hash", dafsa_filename)
+            self.assertEqual(len(responses.calls), 2)
+            self.assertEqual(responses.calls[0].request.method, "POST")
+            self.assertEqual(responses.calls[1].request.method, "PATCH")
```

Comment:
Maybe also assert that POST was on `self.attachment_uri` and PATCH on `self.collection_uri` ?